### PR TITLE
python 3 support

### PR DESCRIPTION
**Pull request by thunterdb**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,31 +2,50 @@ language: scala
...
-    - SPARK_VERSION=1.6.2 SPARK_BUILD="spark-1.6.2-bin-hadoop2.6" SPARK_BUILD_URL="http://d3kbcqa49mib13.cloudfront.net/spark-1.6.2-bin-hadoop2.6.tgz"
+    - PYSPARK_PYTHON=python3 SPARK_VERSION=2.0.1 SPARK_BUILD="spark-2.0.1-bin-hadoop2.7" SPARK_BUILD_URL="http://d3kbcqa49mib13.cloudfront.net/spark-2.0.1-bin-hadoop2.7.tgz"
+    - PYSPARK_PYTHON=python3 SPARK_VERSION=1.6.2 SPARK_BUILD="spark-1.6.2-bin-hadoop2.6" SPARK_BUILD_URL="http://d3kbcqa49mib13.cloudfront.net/spark-1.6.2-bin-hadoop2.6.tgz"
+    - PYSPARK_PYTHON=python2 SPARK_VERSION=2.0.1 SPARK_BUILD="spark-2.0.1-bin-hadoop2.7" SPARK_BUILD_URL="http://d3kbcqa49mib13.cloudfront.net/spark-2.0.1-bin-hadoop2.7.tgz"
+    - PYSPARK_PYTHON=python2 SPARK_VERSION=1.6.2 SPARK_BUILD="spark-1.6.2-bin-hadoop2.6" SPARK_BUILD_URL="http://d3kbcqa49mib13.cloudfront.net/spark-1.6.2-bin-hadoop2.6.tgz"
+
+addons:
+  apt:
+    packages:
+     - python3
```
Comment (mengxr) [inline .travis.yml]: It seems that Python3 should be preinstalled on trusty image: https://docs.travis-ci.com/user/trusty-ci-environment/#Python. Do we need it here?

File: .travis.yml [hunk 2]
```diff
@@ -2,31 +2,50 @@ language: scala
...
+     - python3-pip
 
 matrix:
   exclude:
   - scala: 2.10.5 # Spark 2.0.0 pre-built with Scala 2.11 in the tgz download
-    env: SPARK_VERSION=2.0.1 SPARK_BUILD="spark-2.0.1-bin-hadoop2.7" SPARK_BUILD_URL="http://d3kbcqa49mib13.cloudfront.net/spark-2.0.1-bin-hadoop2.7.tgz"
```
Comment (mengxr) [inline .travis.yml]: Is it simpler if we don't list `scala` as the main language and let the test line be `sbt -Dscala.version=2.10.6`, e.g.?

File: .travis.yml [hunk 3]
```diff
@@ -2,31 +2,50 @@ language: scala
...
   - scala: 2.11.7
-    env: SPARK_VERSION=1.6.2 SPARK_BUILD="spark-1.6.2-bin-hadoop2.6" SPARK_BUILD_URL="http://d3kbcqa49mib13.cloudfront.net/spark-1.6.2-bin-hadoop2.6.tgz"
+    env: PYSPARK_PYTHON=python2 SPARK_VERSION=1.6.2 SPARK_BUILD="spark-1.6.2-bin-hadoop2.6" SPARK_BUILD_URL="http://d3kbcqa49mib13.cloudfront.net/spark-1.6.2-bin-hadoop2.6.tgz"
 
 before_install:
  - ./bin/download_travis_dependencies.sh
 
 install:
  - pip install --user -r ./python/requirements.txt
+ - pip3 install --user -r ./python/requirements.txt
```
Comment (mengxr) [inline .travis.yml]: We might use `pyenv` to switch python version first and make this line the same for py2 and py3.

File: .travis.yml (hunk 2 above)
Comment (mengxr) [inline .travis.yml]: Btw, shall we also bump scala version from `2.10.5` to `2.10.6`?

File: .travis.yml (hunk 1 above)
Comment (thunterdb) [inline .travis.yml]: You are right, I have not tried to remove these flags since moving to trusty.

File: .travis.yml (hunk 2 above)
Comment (thunterdb) [inline .travis.yml]: I am going to try it out.

File: .travis.yml [hunk 4]
```diff
@@ -1,34 +1,31 @@
-language: scala
-scala:
-  - 2.10.5
-  - 2.11.7
 jdk:
   - oraclejdk7 # openJDK crashes sometimes
-sudo: false
+
+sudo: required
```
Comment (mengxr) [inline .travis.yml]: If we no longer need to install `pip`, does it mean we don't need `sudo` either?

Comment (thunterdb) [inline .travis.yml]: We do. trusty is beta, and fails (jvm package not properly installed)

File: .travis.yml (hunk 3 above)
Comment (thunterdb) [inline .travis.yml]: I suggest we keep it this way for now. It just takes a few seconds and it is simpler to install both.

**Comment by nchammas** [comment]
I suppose this addresses #85.

In the future, it would be good to include "Fixes #85" in the PR description so that when the PR gets merged in, the associated issue gets automatically closed and people are notified that their issue is fixed.

**Comment by thunterdb** [comment]
That's a great suggestion. Yes, we should tag the ticket in the title.

**Pull request merged**